app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,52 @@
 package com.lightricks.feedexercise.data
 
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.GetFeedResponse
+import io.reactivex.Completable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+class FeedRepository(private val apiService: FeedApiService, private val db: FeedDatabase) {
+
+    val feedItems: LiveData<List<FeedItem>> =
+        Transformations.map(db.feedDao().getAll()) {
+            Log.d("eilon", "list = $it")
+            it.toFeedItems()
+        }
+
+    fun refresh(): Completable {
+        return apiService.getFeed()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .flatMapCompletable { feedResponse ->
+                handleResponse(feedResponse)
+            }
+    }
+
+    private fun handleResponse(feedResponse: GetFeedResponse): Completable {
+        val feedEntities = feedResponse.templatesMetadata.map {
+            FeedEntity(
+                it.id,
+                FeedApiService.thumbnailURIPrefix + it.templateThumbnailURI,
+                it.isPremium
+            )
+        }
+        return db.feedDao().insertAll(feedEntities)
```

Comment:
Trying to insert the same item more than once will fail. 
You should either delete old items prior to inserting new ones or set a `OnConflictStrategy` for insertion.